internal/replacements/replacements.go

```diff
@@ -228,24 +223,36 @@ func processGroupNeedsRemovalForPod(cluster *fdbv1beta2.FoundationDBCluster, pod
 	}
 
 	if pointer.BoolDeref(cluster.Spec.ReplaceInstancesWhenResourcesChange, false) {
-		desiredSpec, err := internal.GetPodSpec(cluster, processGroupStatus)
-		if err != nil {
-			return false, err
-		}
-
-		if resourcesNeedsReplacement(desiredSpec.Containers, pod.Spec.Containers) {
+		if resourcesNeedsReplacement(spec.Containers, pod.Spec.Containers) {
 			logger.Info("Replace process group",
 				"reason", "Resource requests have changed")
 			return true, nil
 		}
 
-		if resourcesNeedsReplacement(desiredSpec.InitContainers, pod.Spec.InitContainers) {
+		if resourcesNeedsReplacement(spec.InitContainers, pod.Spec.InitContainers) {
 			logger.Info("Replace process group",
 				"reason", "Resource requests have changed")
 			return true, nil
 		}
 	}
 
+	// Some k8s instances have security context vetting which may edit the spec automatically.
+	// This would cause changes to security context on a pod or container
+	// to constantly be seen as having a security context change, hence we want to feature guard this
+	// and also guard on the spec hash below
+	// https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/
+	if replaceOnSecurityContextChange {
+		if pod.ObjectMeta.Annotations[fdbv1beta2.LastSpecKey] == specHash {
+			// no changes have been made outside of server-side injected values, do not check for changes
+			// to avoid looping
+			return false, nil
+		}
+		desiredPod, err := internal.GetPod(cluster, processGroupStatus)
```

Comment:
Is there any need to generate the pod struct? Could we just make use of the already generated `spec`?